frontend/src/components/header/header.jsx

```diff
@@ -1,74 +1,73 @@
-        <LongButton value="Registracija" type="filled" />
-      </Link>
-    </div>
-  );
-}
-
-export const Header = () => {
-  const [toggleState, settoggleState] = useState(true);
-
-  const toggleNav = () => {
-    settoggleState(!toggleState);
-  };
-
-  return (
-    <header className={styles.header}>
-      <div className={styles.imageWrapper}>
-        <Image src="/caregivers.svg" alt="Logo" width={144} height={60} className="image" />
-      </div>
-      <div className={styles.navbar}>
-        {tabItems.map((tab, index) => {
-          return (
-            <div key={index} className={styles.tabItem}>
-              <Link href={tab.pathname} passHref>
-                <span className={styles.tabName}>{tab.name}</span>
-              </Link>
-            </div>
-          );
-        })}
-      </div>
-      <Buttons />
-      <div className={styles.responsiveNav}>
-        {toggleState ? (
-          <FontAwesomeIcon icon={faBars} className={styles.icon} onClick={toggleNav} />
-        ) : (
-          <FontAwesomeIcon icon={faXmark} className={styles.icon} onClick={toggleNav} />
-        )}
-
-        <div className={styles.nav + " " + (!toggleState ? styles.show : null)}>
-          {tabItems.map((tab, index) => {
-            return (
-              <div key={index} className={styles.tabItem}>
-                <Link href={tab.pathname} passHref>
-                  <span className={styles.tabName}>{tab.name}</span>
-                </Link>
-              </div>
-            );
-          })}
-          <Buttons />
-        </div>
-      </div>
-    </header>
-  );
-};
+import React, { useCallback, useEffect, useState } from "react";
+import Image from "next/image";
+import Link from "next/link";
+import { LongButton } from "shared-components/long-button/long.button";
+import { tabItems } from "./tabItems";
+import styles from "./header.module.scss";
+import AuthService from "../../pages/api/authService";
+
+function AuthenticatedMenu({ onLogout }) {
+
+	const logout = useCallback(async () => {
+		await AuthService.logout();
+		onLogout();
+	}, [onLogout]);
+
+	return (
+		<div className={styles.authenticatedMenu}>
+			<Image src="/avatar.svg" alt="avatar" width={65} height={65}/>
+			<div className={styles.logoutButton} onClick={logout}>
+				<span className={styles.tabLabel}>Logout</span>
+			</div>
+		</div>
+	);
+}
+
+function AnonymousMenu() {
+	return (
+		<div className={styles.anonymousMenu}>
+			<Link href="/login" passHref>
+				<LongButton value="Prijava" type="border"/>
+			</Link>
+			<Link href="/registration" passHref>
+				<LongButton value="Registracija" type="filled"/>
+			</Link>
+		</div>
+	);
+}
+
+export const Header = () => {
+	const [isLoadingUser, setIsLoadingUser] = useState(true);
+	const [user, setUser] = useState(null);
+
+	useEffect(() => {
+		if (isLoadingUser) {
+			setUser(AuthService.getUser());
+			setIsLoadingUser(false);
+		}
+	}, [isLoadingUser]);
+
+	if (isLoadingUser) return null;

```

Comment:
Ovde mozemo kasnije dodati neki spiner, sta mislis? Kreiraj issue pa ako stignemo

Reply:
Da, to sam i ja razmisljao. Evo issue: #95